### PR TITLE
tidb: add a new command line flag for redact

### PR DESCRIPTION
**Pull request by xhebox**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): pingcap/tidb#51306

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: command-line-flags-for-tidb-configuration.md [hunk 1]
```diff
@@ -224,6 +224,12 @@ summary: TiDB 配置参数包括启动参数和环境变量。启动参数包括
 + 设置 TiDB server CPU 亲和性，以 "," 逗号分隔，例如 "1,2,3"
 + 默认：""
 
+## `--redact`
+
++ 设置 TiDB server 是否脱敏文件。
++ 默认：false
++ 可以用 `./tidb-server --redact=xxx collect-log <input> <output>` 可以将 `input` 日志文件脱敏或者还原，并输出到 `output`。当值为 `true` 脱敏，否则还原。参考系统变量 [tidb_redact_log](/system-variables.md#tidb-redact-log)。
```
Comment (hfxsd) [inline command-line-flags-for-tidb-configuration.md:231]:
```suggestion
+ 可以使用 `./tidb-server --redact=xxx collect-log <input> <output>` 将 `<input>` 日志文件进行脱敏或者还原，并输出到 `output`。当值为 `true` 时脱敏，为 `false` 时还原。更多详情，请参考系统变量 [tidb_redact_log](/system-variables.md#tidb-redact-log)。
```

File: command-line-flags-for-tidb-configuration.md [hunk 2]
```diff
@@ -224,6 +224,12 @@ summary: TiDB 配置参数包括启动参数和环境变量。启动参数包括
 + 设置 TiDB server CPU 亲和性，以 "," 逗号分隔，例如 "1,2,3"
 + 默认：""
 
+## `--redact`
+
++ 设置 TiDB server 是否脱敏文件。
++ 默认：false
++ 可以使用 `./tidb-server --redact=xxx collect-log <input> <output>` 将 `<input>` 日志文件进行脱敏或者还原，并输出到 `output`。当值为 `true` 时脱敏，为 `false` 时还原。更多详情，请参考系统变量 [tidb_redact_log](/system-variables.md#tidb-redact-log)。
```
Comment (hfxsd) [inline command-line-flags-for-tidb-configuration.md:231]:
```suggestion
+ 可以使用 `./tidb-server --redact=xxx collect-log <input> <output>` 将 `<input>` 日志文件进行脱敏或者还原，并输出到 `<output>`。当值为 `true` 时脱敏，为 `false` 时还原。更多详情，请参考系统变量 [tidb_redact_log](/system-variables.md#tidb-redact-log)。
```

File: command-line-flags-for-tidb-configuration.md [hunk 3]
```diff
@@ -224,6 +224,12 @@ summary: TiDB 配置参数包括启动参数和环境变量。启动参数包括
 + 设置 TiDB server CPU 亲和性，以 "," 逗号分隔，例如 "1,2,3"
 + 默认：""
 
+## `--redact`
+
++ 设置 TiDB server 是否脱敏文件。
++ 默认：false
++ 可以使用 `./tidb-server --redact=xxx collect-log <input> <output>` 将 `<input>` 日志文件进行脱敏或者还原，并输出到 `<output>`。当值为 `true` 时脱敏，为 `false` 时还原。更多详情，请参考系统变量 [tidb_redact_log](/system-variables.md#tidb-redact-log)。
```
Comment (hfxsd) [inline command-line-flags-for-tidb-configuration.md:231]:
```suggestion
+ 可以使用 `./tidb-server --redact=xxx collect-log <input> <output>` 将 `<input>` 日志文件进行脱敏或者还原，并输出到 `<output>`。当值为 `true` 时脱敏，为 `false` 时还原。更多详情，请参考系统变量 [`tidb_redact_log`](/system-variables.md#tidb-redact-log)。
```

File: command-line-flags-for-tidb-configuration.md [hunk 4]
```diff
@@ -224,6 +224,12 @@ summary: TiDB 配置参数包括启动参数和环境变量。启动参数包括
 + 设置 TiDB server CPU 亲和性，以 "," 逗号分隔，例如 "1,2,3"
 + 默认：""
 
+## `--redact`
+
++ 设置 TiDB server 是否脱敏文件。
++ 默认：false
++ 可以使用 `./tidb-server --redact=xxx collect-log <input> <output>` 将 `<input>` 日志文件进行脱敏或者还原，并输出到 `<output>`。当值为 `true` 时脱敏，为 `false` 时还原。更多详情，请参考系统变量 [`tidb_redact_log`](/system-variables.md#tidb-redact-log)。
```
Comment (hfxsd) [inline command-line-flags-for-tidb-configuration.md:231]:
```suggestion
+ 可以使用 `./tidb-server --redact=xxx collect-log <input> <output>` 将 `<input>` 日志文件进行脱敏或者还原，并输出到 `<output>`。当值为 `true` 时脱敏，为 `false` 时还原。更多详情，请参考系统变量 [`tidb_redact_log`](/system-variables.md#tidb_redact_log)。
```

File: command-line-flags-for-tidb-configuration.md [hunk 5]
```diff
@@ -224,6 +224,12 @@ summary: TiDB 配置参数包括启动参数和环境变量。启动参数包括
 + 设置 TiDB server CPU 亲和性，以 "," 逗号分隔，例如 "1,2,3"
 + 默认：""
 
+## `--redact`
+
++ 设置 TiDB server 是否脱敏文件。
```
Comment (hfxsd) [inline command-line-flags-for-tidb-configuration.md:229]:
```suggestion
+ 设置 TiDB server 是否脱敏日志文件。
```

File: command-line-flags-for-tidb-configuration.md [hunk 6]
```diff
@@ -224,6 +224,12 @@ summary: TiDB 配置参数包括启动参数和环境变量。启动参数包括
 + 设置 TiDB server CPU 亲和性，以 "," 逗号分隔，例如 "1,2,3"
 + 默认：""
 
+## `--redact`
+
++ 设置 TiDB server 是否脱敏日志文件。
++ 默认：false
++ 可以使用 `./tidb-server --redact=xxx collect-log <input> <output>` 将 `<input>` 日志文件进行脱敏或者还原，并输出到 `<output>`。当值为 `true` 时脱敏，为 `false` 时还原。更多详情，请参考系统变量 [`tidb_redact_log`](/system-variables.md#tidb_redact_log)。
```
Comment (Defined2014) [inline command-line-flags-for-tidb-configuration.md:231]:
```suggestion
+ 值可以为 `true` 或者 `false`，当值为 `true` 时为脱敏操作，为 `false` 时为还原操作，具体使用方法为执行 `./tidb-server --redact=xxx collect-log <input> <output>` 将 `<input>` 日志文件进行脱敏或者还原，并输出到 `<output>`。更多详情，请参考系统变量 [`tidb_redact_log`](/system-variables.md#tidb_redact_log)。
```

**Review by Defined2014 — APPROVED** [review]

File: command-line-flags-for-tidb-configuration.md [hunk 7]
```diff
@@ -224,6 +224,12 @@ summary: TiDB 配置参数包括启动参数和环境变量。启动参数包括
 + 设置 TiDB server CPU 亲和性，以 "," 逗号分隔，例如 "1,2,3"
 + 默认：""
 
+## `--redact`
+
++ 设置 TiDB server 是否在使用子命令 `collect-log` 时脱敏日志文件。
++ 默认：false
++ 值可以为 `true` 或者 `false`。当值为 `true` 时为脱敏操作，所有 `‹..›` 标记的字段被替换为 `?`；`false` 时为还原操作，所有标记符号被去除。具体使用方法为执行 `./tidb-server --redact=xxx collect-log <input> <output>` 将 `<input>` 指向的 TiDB server 日志文件进行脱敏或者还原，并输出到 `<output>`。更多详情，请参考系统变量 [`tidb_redact_log`](/system-variables.md#tidb_redact_log)。
```
Comment (hfxsd) [inline command-line-flags-for-tidb-configuration.md:231]:
```suggestion
+ 取值为 `true` 时为脱敏操作，所有 `‹..›` 标记的字段被替换为 `?`。取值为 `false` 时为还原操作，所有标记符号被去除。具体使用方法为，执行 `./tidb-server --redact=xxx collect-log <input> <output>` 将 `<input>` 指向的 TiDB server 日志文件进行脱敏或者还原，并输出到 `<output>`。更多详情，请参考系统变量 [`tidb_redact_log`](/system-variables.md#tidb_redact_log)。
```

**Review by hfxsd — APPROVED** [review]

File: command-line-flags-for-tidb-configuration.md [hunk 8]
```diff
@@ -224,6 +224,12 @@ summary: TiDB 配置参数包括启动参数和环境变量。启动参数包括
 + 设置 TiDB server CPU 亲和性，以 "," 逗号分隔，例如 "1,2,3"
 + 默认：""
 
+## `--redact`
+
++ 设置 TiDB server 是否在使用子命令 `collect-log` 时脱敏日志文件。
++ 默认：false
++ 取值为 `true` 时为脱敏操作，所有 `‹..›` 标记的字段被替换为 `?`。取值为 `false` 时为还原操作，所有标记符号被去除。具体使用方法为，执行 `./tidb-server --redact=xxx collect-log <input> <output>` 将 `<input>` 指向的 TiDB server 日志文件进行脱敏或者还原，并输出到 `<output>`。更多详情，请参考系统变量 [`tidb_redact_log`](/system-variables.md#tidb_redact_log)。
```
Comment (hfxsd) [inline command-line-flags-for-tidb-configuration.md:231]:
```suggestion
+ 取值为 `true` 时为脱敏操作，所有被标记符号 `‹ ›` 包裹的字段会被替换为 `?`。取值为 `false` 时为还原操作，所有标记符号被去除。具体使用方法为，执行 `./tidb-server --redact=xxx collect-log <input> <output>` 将 `<input>` 指向的 TiDB server 日志文件进行脱敏或者还原，并输出到 `<output>`。更多详情，请参考系统变量 [`tidb_redact_log`](/system-variables.md#tidb_redact_log)。
```

File: command-line-flags-for-tidb-configuration.md [hunk 9]
```diff
@@ -224,6 +224,12 @@ summary: TiDB 配置参数包括启动参数和环境变量。启动参数包括
 + 设置 TiDB server CPU 亲和性，以 "," 逗号分隔，例如 "1,2,3"
 + 默认：""
 
+## `--redact`
+
++ 设置 TiDB server 是否在使用子命令 `collect-log` 时脱敏日志文件。
++ 默认：false
++ 取值为 `true` 时为脱敏操作，所有被标记符号 `‹ ›` 包裹的字段会被替换为 `?`。取值为 `false` 时为还原操作，所有标记符号会被去除。具体使用方法为，执行 `./tidb-server --redact=xxx collect-log <input> <output>` 将 `<input>` 指向的 TiDB server 日志文件进行脱敏或者还原，并输出到 `<output>`。更多详情，请参考系统变量 [`tidb_redact_log`](/system-variables.md#tidb_redact_log)。
```
Comment (lilin90) [inline command-line-flags-for-tidb-configuration.md:231]:
```suggestion
+ 取值为 `true` 时为脱敏操作，所有被标记符号 `‹ ›` 包裹的字段会被替换为 `?`。取值为 `false` 时为还原操作，所有标记符号会被去除。具体使用方法为：执行 `./tidb-server --redact=xxx collect-log <input> <output>` 将 `<input>` 指向的 TiDB server 日志文件进行脱敏或者还原，并输出到 `<output>`。更多详情，请参考系统变量 [`tidb_redact_log`](/system-variables.md#tidb_redact_log)。
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**